02.calendar/calendar.rb

```diff
@@ -0,0 +1,54 @@
+require 'optparse'
+require 'date'
+params = ARGV.getopts("m:","y:")
+
+month = params["m"].to_i
+year = params["y"].to_i
+
+def check_month(month)
+  if month == 0
+    return Date.today.month
+  else
+    return month
+  end
+end
+
+def check_year(year)
+  if year == 0
+    return Date.today.year
+  else
+    return year
+  end
+end
+
+def days_in_month(month, year)
+  if month == 1 || month == 3 || month == 5 || month == 7 || month == 8 || month == 10 || month == 12
+    return 31
+  elsif month == 4 || month == 6 || month == 9 || month == 11
+    return 30
+  elsif Date.leap?(year)
+    return 29
+    else return 28
+  end
+end
+
+_month = check_month(month)
+_year = check_year(year)
```

Comment:
`_`で始める変数名はライブラリやフレームワークの中で特別な意味合いを持つことも多いので、一般的には使用を避けた方がいいと思います。シンプルに`month`, `year`を上書きしてしまえばいいと思います。

```suggestion
month = check_month(month)
year = check_year(year)
```